felix/ipsets/ipsets.go

```diff
@@ -791,6 +864,8 @@ func (s *IPSets) writeUpdates(setName string, w io.Writer) (err error) {
 // around the fact that we sometimes see transient failures to remove
 // temporary IP sets.
 func (s *IPSets) nextFreeTempIPSetName() string {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
```

Comment:
I think the contention is very low on this lock since the background goroutines spend most of their time doing the IO, which is done without the locks held.  I think it's best to keep the simple coarse lock rather than optimise it away because the lock operations only take usec vs the IO, which takes millis to seconds.